src/shaka_scheme/runtime/stdproc/equivalence_predicates.cpp

```diff
@@ -0,0 +1,181 @@
+//
+// Created by Sean Takafuji on 2/23/2018.
+//
+
+#include "shaka_scheme/runtime/stdproc/equivalence_predicates.hpp"
+#include "shaka_scheme/system/core/lists.hpp"
+
+namespace shaka {
+namespace stdproc {
+namespace impl {
+
+Args is_eqv(Args args) {
+  if (args.size() != 2) {
+    throw InvalidInputException(1,
+                                "eqv?: Invalid number of arguments for procedure");
+  }
+
+  auto left_type = args[0]->get_type();
+  auto right_type = args[1]->get_type();
+
+  // If the Base Data types aren't equivalent, return false;
+  if (left_type != right_type) {
+    return {create_node(shaka::Data(shaka::Boolean(false)))};
+  }
+  auto result = false;
+
+  // Data Type: Boolean
+  if (left_type == Data::Type::BOOLEAN) {
+    result = args[0]->get<Boolean>() == args[1]->get<Boolean>();
+  }
+    // Data Type: Symbol
+  else if (left_type == Data::Type::SYMBOL) {
+    result = args[0]->get<Symbol>() == args[1]->get<Symbol>();
+  }
+    // Data Type: Number
+  else if (left_type == Data::Type::NUMBER) {
+    // Check if the Number type is the same (Integer, Real, Rational)
+    auto number_left_type = args[0]->get<Number>().get_type();
+    auto number_right_type = args[1]->get<Number>().get_type();
+    if (number_left_type != number_right_type) {
+      return {create_node(shaka::Data(shaka::Boolean(result)))};
+    }
+    result = args[0]->get<Number>() == args[1]->get<Number>();
+  }
+    // Data Type: String
+  else if (left_type == Data::Type::STRING) {
+    result = args[0]->get<String>() == args[1]->get<String>();
+  }
+    // Data Type: Null List
+  else if (left_type == Data::Type::NULL_LIST) {
+    result = true;
+  }
```

Comment:
Missing case for `DATA_PAIR`

Reply:
Also, `eq?` needs to be able to determine if two procedures refer to the same procedure. I believe you can do all of these cases by comparing the pointers' internal values themselves (try use `get()` on the `std::shared_ptr<Data>`).